### ISSUE TITLE
check failures using data.table from github master

### ISSUE THREAD

**Issue by tdhock** [nicolarighetti/CooRTweet#16]
Hi @nicolarighetti 
After installing data.table from github master,
```r
data.table::update_dev_pkg()
```
And then running R CMD check on CooRTweet, I get the following new failures, (which are not present if you use data.table from CRAN)
```
* checking tests ...
 ERROR
Running the tests in 'tests/testthat.R' failed.
Last 13 lines of output:
   3.     +-data.table::melt(...)
   4.     \-data.table::melt.data.table(...)
  -- Error ('test-simulated_data.R:220:5'): larger time window -------------------
  Error in `melt.data.table(x, id.vars = c("object_id", "time_delta"), measure.vars = patterns("^content_id", 
      "^id_user"), value.name = c("content_id", "id_user"))`: When 'measure.vars' is a list, 'value.name' must be a character vector of length =1 or =length(measure.vars).
  Backtrace:
      x
   1. \-CooRTweet (local) sim_test(...) at test-simulated_data.R:220:4
   2.   \-CooRTweet::group_stats(result) at test-simulated_data.R:48:4
   3.     +-data.table::melt(...)
   4.     \-data.table::melt.data.table(...)
  
  [ FAIL 9 | WARN 0 | SKIP 0 | PASS 33 ]
  Error: Test failures
  Execution halted
...

* checking re-building of vignette outputs ... ERROR
Error(s) in re-building vignettes:
  ...
--- re-building 'vignette.Rmd' using rmarkdown

Quitting from lines 103-105 [unnamed-chunk-4] (vignette.Rmd)
Error: processing vignette 'vignette.Rmd' failed with diagnostics:
When 'measure.vars' is a list, 'value.name' must be a character vector of length =1 or =length(measure.vars).
--- failed re-building 'vignette.Rmd'

SUMMARY: processing the following file failed:
  'vignette.Rmd'

Error: Vignette re-building failed.
Execution halted
```
data.table needs to ensure that updates do not break dependent packages like CooRTweet, before uploading new versions to CRAN, so can you please submit an updated version of CooRTweet to CRAN, that fixes this test failure? In particular, I would suggest to remove the definition of `patterns = function(...)NULL` inside of `group_stats`, which would fix these failures. (the new github master version of data.table melt takes patterns from user code if it is defined, whereas the previous CRAN version of data.table melt would always use data.table::patterns)

**Comment by mrwunderbar666** [comment]
Thank you @tdhock for submitting this issue. I finally had time to look into it, but it turns out that data.table still does not export `patterns`

Using your proposed changes, the tests pass, but checks fail (which will lead to rejection when submitting to CRAN):

```
❯ checking R code for possible problems ... NOTE
  generate_network: no visible global function definition for ‘patterns’
  group_stats: no visible global function definition for ‘patterns’
  user_stats: no visible global function definition for ‘patterns’
  Undefined global functions or variables:
    patterns
```

Using the recent github version of data.table:
```r
> data.table::update_dev_pkg()
R data.table package is up-to-date at 88039186915028ab3c93ccfd8e22c0d1c3534b1a (1.14.9)
```

I would recommend properly exporting `patterns` from data.tables so that dependend packages don't have to find hacks to pass CRAN checks

**Comment by tdhock** [comment]
use patterns = NULL please

**Comment by tdhock** [comment]
use patterns = NULL please

**Comment by tdhock** [comment]
This issue is now fixed, as of CooRTweet 1.5.0 on CRAN, 5 Sep 2023.
Thanks very much!
